pkg/services/generic.go

```diff
@@ -333,3 +344,66 @@ func (s *sqlGenericService) treeWalkForSqlizer(listCtx *listContext, tslTree tsl
 
 	return tslTree, sqlizer, nil
 }
+
+func isJSONBSearch(search string) bool {
+	if strings.Contains(search, "->>") || strings.Contains(search, "@>") {
+		return true
+	}
+	return false
+}
+
+func validateJSONBSearch(search string) error {
+	// handle ->> search. for example: payload -> 'data' -> 'manifest' -> 'metadata' -> 'labels' ->> 'foo' = 'bar'
+	if strings.Contains(search, "->>") {
+		jsonbStr := stringSplitwithTrim(search, "->>")
+		if err := isValidFields(stringSplitwithTrim(jsonbStr[0], "->")); err != nil {
+			return fmt.Errorf("the search field name is invalid: %v", err)
+		}
+		labelStr := stringSplitwithTrim(jsonbStr[1], "=")
+		if err := validation.IsQualifiedName(labelStr[0]); err != nil {
```

Comment:
ditto, do we ensure labelStr is alway not nil?

Reply:
what is there is no "="?

Reply:
refer to this test
```
		{
			"name":   "complex labels",
			"search": "payload -> 'data' -> 'manifest' -> 'metadata' -> 'labels' ->> 'example.com/version'",
		},
```